tests/integration/components/sl-calendar-test.js

```diff
@@ -0,0 +1,773 @@
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+            viewMode="years"
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        '2020-2029',
+        'The current Decade is set correctly'
+    );
+
+    this.$( '.sl-calendar .prev' ).click();
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        '2010-2019',
+        'The previous Decade is set correctly'
+    );
+});
+
+test( 'When Locked, interacting with the view is not Possible', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 8;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.strictEqual(
+        this.$().hasClass( 'sl-calendar-locked' ),
+        false,
+        'Default rendered component does not have class "sl-calendar-locked"'
+    );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            locked=true
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.ok(
+        this.$( '.sl-calendar' ).hasClass( 'sl-calendar-locked' ),
+        'Locked, rendered component has class "sl-calendar-locked"'
+    );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'August ' + currentYear,
+        'The current month is set correctly'
+    );
+
+    this.$( '.sl-calendar .next' ).click();
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'August ' + currentYear,
+        'The next month is set correctly'
+    );
+
+    this.$( '.sl-calendar .prev' ).click();
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'August ' + currentYear,
+        'The next month is set correctly'
+    );
+
+    this.$( '.sl-calendar .datepicker-switch' ).click();
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'August ' + currentYear,
+        'The next month is set correctly'
+    );
+});
+
+test( 'Navigating from Month to Year', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
```

Comment:
What role does this serve in this test? The component will still render as expected, especially within the context of this test, without this.